hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/extdataset/ExternalDatasetImpl.java

```diff
@@ -53,6 +53,7 @@ public class ExternalDatasetImpl implements FsDatasetSpi<ExternalVolumeImpl> {
   private final DatanodeStorage storage = new DatanodeStorage(
       DatanodeStorage.generateUuid(), DatanodeStorage.State.NORMAL,
       StorageType.DEFAULT);
+  private long lastDirScannerFinishTime;
```

Comment:
A lot of functions in this class are not implemented. I think we should following the same pattern here. 

getLastDirScannerFinishTime() -> return 0.
setLastDirScannerFinishTime() -> do nothing.

Below are functions related with Trash. 
```
  @Override
  public void enableTrash(String bpid) {

  }

  @Override
  public void clearTrash(String bpid) {

  }

  @Override
  public boolean trashEnabled(String bpid) {
    return false;
  }
```

`SimulatedFSDataset.java `has a different implementation for enableTrash(). So, I guess we don't have to enforce the same implementation between `SimulatedFSDataset` and `ExternalDatasetImpl`. I am fine with the current change.
```
@Override
  public void enableTrash(String bpid) {
    throw new UnsupportedOperationException();
  }
```